src/main/java/com/palantir/stash/stashbot/managers/JenkinsManager.java

```diff
@@ -363,8 +364,19 @@ public Void perform() throws Exception {
     public void triggerBuild(final Repository repo, final JobType jobType,
         final PullRequest pr) {
 
-        final String username = um.getRemoteUser().getUsername();
-        final ApplicationUser su = us.findUserByNameOrEmail(username);
+        ApplicationUser su;
```

Comment:
This looks like a perfectly cromulent solution.  The PR author is as good a stand-in as any.